pkg/integration_test.go

```diff
@@ -4,468 +4,63 @@
-	in.Fields["model"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "facebook/bart-large-cnn"}}
-	op, err := hfCon.Execute([]*structpb.Struct{&in})
-	fmt.Printf("\n op :%v, err:%s", op, err)
-}
+	in, err := base.ConvertToStructpb(openai.TextCompletionInput{
+		Prompt: "how are you doing?",
+		Model:  "gpt-3.5-turbo",
+	})
+	c.Assert(err, qt.IsNil)
 
-func TestHuggingFaceTextClassification(t *testing.T) {
-	req := huggingface.TextClassificationRequest{Inputs: "I like you. I love you"}
-	var in structpb.Struct
-	b, _ := json.Marshal(req)
-	protojson.Unmarshal(b, &in)
-	in.Fields["task"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "TEXT_CLASSIFICATION"}}
-	in.Fields["model"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "distilbert-base-uncased-finetuned-sst-2-english"}}
-	op, err := hfCon.Execute([]*structpb.Struct{&in})
-	fmt.Printf("\n op :%v, err:%s", op, err)
-}
+	logger := zap.NewNop()
+	conn := Init(logger, emptyOptions)
 
-func TestHuggingFaceTextGeneration(t *testing.T) {
-	req := huggingface.TextClassificationRequest{Inputs: "The answer to the universe is"}
-	var in structpb.Struct
-	b, _ := json.Marshal(req)
-	protojson.Unmarshal(b, &in)
-	in.Fields["task"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "TEXT_GENERATION"}}
-	in.Fields["model"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "gpt2"}}
-	op, err := hfCon.Execute([]*structpb.Struct{&in})
-	fmt.Printf("\n op :%v, err:%s", op, err)
-}
+	def, err := conn.GetConnectorDefinitionByID("openai")
+	c.Assert(err, qt.IsNil)
 
-func TestHuggingFaceTokenClassification(t *testing.T) {
-	req := huggingface.TextClassificationRequest{Inputs: "My name is Sarah Jessica Parker but you can call me Jessica"}
-	var in structpb.Struct
-	b, _ := json.Marshal(req)
-	protojson.Unmarshal(b, &in)
-	in.Fields["task"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "TOKEN_CLASSIFICATION"}}
-	in.Fields["model"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "dbmdz/bert-large-cased-finetuned-conll03-english"}}
-	op, err := hfCon.Execute([]*structpb.Struct{&in})
-	fmt.Printf("\n op :%v, err:%s", op, err)
-}
+	uid, err := uuid.FromString(def.GetUid())
+	c.Assert(err, qt.IsNil)
 
-func TestHuggingFaceTranslation(t *testing.T) {
-	req := huggingface.TranslationRequest{Inputs: "Меня зовут Вольфганг и я живу в Берлине"}
-	var in structpb.Struct
-	b, _ := json.Marshal(req)
-	protojson.Unmarshal(b, &in)
-	in.Fields["task"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "TRANSLATION"}}
-	in.Fields["model"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: "Helsinki-NLP/opus-mt-ru-en"}}
-	op, err := hfCon.Execute([]*structpb.Struct{&in})
-	fmt.Printf("\n op :%v, err:%s", op, err)
-}
+	exec, err := conn.CreateExecution(uid, "TASK_TEXT_GENERATION", config, logger)
```

Comment:
❓ Does it make sense to pass the logger when `conn` already has an exported logger? I don't think we'll ever have a use case where we'll want different loggers.

I kept the signature because it's aligned with the other `CreateExecution` statements.

Reply:
I think we can remove all the `logger` argument from `CreateExecution`, we can reuse the logger defined [here](https://github.com/instill-ai/component/blob/c139af45d2a7f6e4117e8ec8ac355cac207630b5/pkg/base/component.go#L75)

Reply:
👍 This requires modifying [the `IComponent` interface](https://github.com/instill-ai/component/blob/main/pkg/base/component.go#L33). As a refactorisation task that touches several repos, I'd go for a separate task (along with the comment below). I updated the linear ticket to reflect this.

Reply:
No problem